### ISSUE TITLE
Send back refreshed nonce when updating

### ISSUE THREAD

**Issue by westonruter** [xwp/wp-widget-customizer#4]

**Comment by bobbravo2** [comment]
The nonce is valid for that action for 12 hours - so if we refresh it - we'd need to add a wp_create_nonce("action_$index") to make it unique / expire. That way the nonce would only be valid for that request. 

The reason I'm unsure - is that my understanding of NONCE's - is that the original NONCE would still be valid - therefore we'd only be adding a layer of complexity by refreshing the nonce. If a user's wp_auth_cookie is compromised, they'd still be able to preview the customizer / perform that action.

Thoughts?

**Comment by westonruter** [comment]
Yes, multiple nonces would continue to be valid concurrently. But it's not a problem to have more than once nonce that is valid at the same time. My concern is that someone could have the customizer left open in a window over night, and then when they come back and try to make a change, they could get an error because of an invalid nonce. But I suppose just refreshing the nonce won't do it either, as if they return to their browser after 24 hours, the last nonce they have would have been long expired anyway. Hummm.

**Comment by bobbravo2** [comment]
Yeah. I think after a night left on the customizer their WordPress session would have expired as well. Perhaps this issue was already dealt with in the customizer - and it'd catch an invalid nonce after trying to refresh the preview - and then trigger a full page refresh to get new nonces?

**Comment by c3mdigital** [comment]
The Heartbeat API might be something to look into as it allows you to login without a page refresh.

**Comment by westonruter** [comment]
Just [discovered](https://twitter.com/westonruter/status/394214714214858752) something very useful for this: [`wp_verify_nonce()`](https://github.com/WordPress/WordPress/blob/fde786ab5d687a79ceeeeacf677100b9f88280d9/wp-includes/pluggable.php#L1291-L1307) returns `1` if the supplied nonce was generated 0-12 hours ago, and it returns `2` if it was generated 12-24 hours ago.
